pkg/oc/util/dir/util.go

```diff
@@ -0,0 +1,28 @@
+package dir
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+// ConfigDir provides directory creation for cluster up components. In case the hostDir is specified, it will use
+// that directory as a base directory for all configs. In case the hostDir already exists, this will return true.
+// In case it is not specified a temporary directory is created.
+func ConfigDir(hostDir, component string) (string, error) {
```

Comment:
This is bounded to cluster-up.  Never auto create anything from this method.  If they mess up, just fail.